enterprise/aibridgeproxyd/aibridgeproxyd.go

```diff
@@ -173,17 +180,28 @@ func (s *Server) Close() error {
 // loadMitmCertificate loads the CA certificate and private key for MITM proxying.
 // This function is safe to call concurrently - the certificate is only loaded once
 // into the global goproxy.GoproxyCa variable.
-func loadMitmCertificate(certFile, keyFile string) error {
+// Returns the PEM-encoded certificate for serving to clients.
+func loadMitmCertificate(certFile, keyFile string) ([]byte, error) {
 	tlsCert, err := tls.LoadX509KeyPair(certFile, keyFile)
 	if err != nil {
-		return xerrors.Errorf("load CA certificate: %w", err)
+		return nil, xerrors.Errorf("load CA certificate: %w", err)
+	}
+
+	if len(tlsCert.Certificate) == 0 {
```

Comment:
Should we add a TODO here to validate whether the proxy CA is always on index 0? And link it to issue: https://github.com/coder/internal/issues/1204